contracts/lending/ttoken/strategies/compound/TTokenCompoundStrategy_2.sol

```diff
@@ -0,0 +1,205 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+// contracts
+import {
+    RolesMods
+} from "../../../../contexts2/access-control/roles/RolesMods.sol";
+import { ADMIN } from "../../data.sol";
+
+// Interfaces
+import { ICErc20 } from "../../../../shared/interfaces/ICErc20.sol";
+import { TTokenStrategy } from "../TTokenStrategy.sol";
+import { ITToken_V3 } from "../../ITToken_V3.sol";
+
+// Libraries
+import {
+    SafeERC20
+} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { NumbersLib } from "../../../../shared/libraries/NumbersLib.sol";
+import { LibMeta } from "../../../../shared/libraries/LibMeta.sol";
+import {
+    LibCompound
+} from "../../../../escrow/dapps/libraries/LibCompound.sol";
+
+// Storage
+import "../../token-storage.sol" as TokenStorage;
+import "./compound-storage.sol" as CompoundStorage;
+
+contract TTokenCompoundStrategy_2 is RolesMods, TTokenStrategy {
```

Comment:
In order to showcase the fact that we are only modifying the behavior of `totalUnderlyingSupply`, lets extend from `TTokenCompoundStrategy` and only modify the `totalUnderlyingSupply` function and `constructor`.

By doing this, in the `totalUnderlyingSupply` function you can simply do this:
```solidity
        uint256 currentSupply = super.totalUnderlyingSupply();
```
instead of:
```solidity
        uint256 currentSupply = tokenStore().underlying.balanceOf(
            address(this)
        ) + compoundStore().cToken.balanceOfUnderlying(address(this));
```